az-reportal/src/main/java/azkaban/viewer/reportal/ReportalServlet.java

```diff
@@ -1045,7 +1045,19 @@ private String validateAndSaveReport(final HttpServletRequest req,
     page.add("projectId", project.getId());
 
     try {
-      report.createZipAndUpload(projectManager, user, this.reportalStorageUser);
+      // Fetch the uploader's IP
+      String uploaderIPAddr = "";
+      if (req != null) {
+        uploaderIPAddr = req.getHeader("X-FORWARDED-FOR");
+        if (uploaderIPAddr == null || "".equals(uploaderIPAddr)) {
```

Comment:
.isEmpty() or Apache Commons StringUtils.isEmpty(str) or Guava's Strings.isNullOrEmpty if any of those are available?

Reply:
Copy paste hazard. Makes sense!